### PR TITLE
imgui: New port

### PR DESCRIPTION
**Pull request by judaew**
Version 1.83

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.5.1 20G80 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/imgui/Portfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+checksums           rmd160 ce2df5295f32903d58d9ef4b2378147ddc362fec \
+                    sha256 3de7af5078882edbebff452f2a9ee362ae05107896e14f6ab33621a3473eb95a \
+                    size 1448525
+
+post-extract {
+    # https://raw.githubusercontent.com/microsoft/vcpkg/master/ports/imgui/CMakeLists.txt
+    # https://raw.githubusercontent.com/microsoft/vcpkg/master/ports/imgui/imgui-config.cmake.in
+    copy ${filespath}/CMakeLists.txt ${worksrcpath}
+    copy ${filespath}/imgui-config.cmake.in ${worksrcpath}
+}
```
Comment (reneeotten) [inline devel/imgui/Portfile:39]: why is upstream not providing these files if they're needed? Isn't it a bit unconventional that you need to grab installation/build files from a "random" repository?

**Comment by judaew** [comment]
> why is upstream not providing these files if they're needed? Isn't it a bit unconventional that you need to grab installation/build files from a "random" repository?

This project, although it can be built as a library, for which it specifies in its wiki several CMakeLists.txt, itself recommends connecting by adding a cpp file to the project. 

For this reason, I will close PR and add the port to my repository.

**Comment by reneeotten** [comment]
> For this reason, I will close PR and add the port to my repository.

that's probably not needed. I was just wondering out loud why this was needed - the fact that I didn't understand it (without looking carefully) doesn't mean there is something wrong with it ;)

**Comment by judaew** [comment]
I try to add new projects to MacPorts to help other users and because I like doing it.

The `imgui` project is the one I'm learning to work with programming in gamedev. I first created the port in the hope that it might be of use to someone else. But after getting to know the project better I understand the reason why it doesn't provide CMakeLists.txt by default. To quote the author from one PR:

> I understand why cmake is used. I personally think it's unnecessary for imgui as the point of the library is that you can register the .cpp file in your application and I encourage you to do so (you can add imgui cpp files from your main app cmake file).
> Only one install ImGui package to your environment and use it anyware. It is a little harmful as it prevents people from configuring ImGui for their application via imconfig.h. While people are free to repackage imgui if they like I think we shouldn't encourage it. ImGui is designed to be compiled in your engine/application and it is easy to compile (the absence of build files is an indicator that it doesn't need anything to be built).

On this explanation of the author of the project, I realized that the port is unnecessary.

**Pull request closed without merging**